src/main/kotlin/com/ajax/cryptocurrency/bpp/ScheduledBackgroundJobBeanPostProcessor.kt

```diff
@@ -0,0 +1,49 @@
+package com.ajax.cryptocurrency.bpp
+
+import com.ajax.cryptocurrency.annotation.ScheduledBackgroundJobStarter
+import com.ajax.cryptocurrency.parser.interfaces.ParserInterface
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import kotlin.reflect.KClass
+
+@Component
+class ScheduledBackgroundJobBeanPostProcessor(
+    @Value("\${cryptocurrency.name}") val cryptocurrencyNames: List<String>,
+) : BeanPostProcessor {
+    private val beans = mutableMapOf<String, KClass<*>>()
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        val beanClass = bean::class
+        beanClass.java.getAnnotation(ScheduledBackgroundJobStarter::class.java)?.let {
+            beans[beanName] = beanClass
+        }
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
+        val originalBean = beans[beanName]
+        if (originalBean != null) {
+            val annotation = originalBean.java.getAnnotation(ScheduledBackgroundJobStarter::class.java)
+            val scheduler = Executors.newScheduledThreadPool(cryptocurrencyNames.size)
```

Comment:
Sorry, forgot about one more thing.

Suppose you have multiple beans that are annotated with `ScheduledBackgroundJobStarter`.
So, for each such bean you create personal executor service.

It's better to use common executor for all such beans.
1. declare scheduler as property of the BPP
```kotlin
class ScheduledBackgroundJobBeanPostProcessor(...) {
    private val scheduler = Executors.newScheduledThreadPool(cryptocurrencyNames.size)
    
    ...
}    
```
2. define pre destroy function to shutdown the executor:
```kotlin
@PreDestroy
private fun shutdown() {
   executor.shutdownNow()
}
```

Reply:
done